gen/app-gen

```diff
@@ -0,0 +1,307 @@
+                echo "ERROR: application name must contain only alpha-numerics, _ or -"
+                show_help_and_exit_error
+            fi
+            application_name="$2"
+            shift 2
+            ;;
+        --manifests-dir | -r)
+            if [ -z "$2" ]; then
+                show_help_and_exit_error
+            fi
+            manifests_dir="$2"
+            shift 2
+            ;;
+        --orchestrator | -r)
+            if [ -z "$2" ]; then
+                show_help_and_exit_error
+            fi
+            orchestrator="$2"
+            shift 2
+            ;;
+        --platform | -p)
+            if [ -z "$2" ]; then
+                show_help_and_exit_error
+            fi
+            platform="$2"
+            shift 2
+            ;;
+        --image | -i)
+            # --image docker.io/library/debian:11,docker.io/library/debian:latest@sha256:a94cd7c7d58f483affd5937853ad4d24caa18cd7c2ec9ef65a9e528dfbc5eb07 --image docker.io/library/postgres:15.1
+            if [ -z "$2" ]; then
+                show_help_and_exit_error
+            fi
+            images+=($(echo "$2" | cut -f1 -d,)) # current
+            images+=($(echo "$2" | cut -f2 -d,)) # new, if current!=new then we need to generate delta.
+            shift 2
+            ;;
+        --device-type | -t)
+            if [ -z "$2" ]; then
+                show_help_and_exit_error
+            fi
+            device_types="$device_types $1 $2"
+            shift 2
+            ;;
+        --artifact-name | -n)
+            if [ -z "$2" ]; then
+                show_help_and_exit_error
+            fi
+            artifact_name=$2
+            shift 2
+            ;;
+        --output-path | -o)
+            if [ -z "$2" ]; then
+                show_help_and_exit_error
+            fi
+            output_path=$2
+            shift 2
+            ;;
+        -h | --help)
+            show_help
+            exit 0
+            ;;
+        --)
+            passthrough=1
+            shift
+            ;;
+        -*)
+            echo "Error: unsupported option $1"
+            show_help_and_exit_error
+            ;;
+        *)
+            shift
+            ;;
+    esac
+done
+
+if [ -z "${artifact_name}" ]; then
+    echo "Artifact name not specified. Aborting."
+    show_help_and_exit_error
+fi
+
+if [ -z "${device_types}" ]; then
+    echo "Device type not specified. Aborting."
+    show_help_and_exit_error
+fi
+
+if [ -z "${orchestrator}" ]; then
+    echo "Orchestrator not specified. Aborting."
+    show_help_and_exit_error
+fi
+
+if [ -z "${platform}" ]; then
+    echo "Platform not specified. Aborting."
+    show_help_and_exit_error
+fi
+
+if [ -z "${manifests_dir}" ]; then
+    echo "Directory containing manifests not specified. Aborting."
+    show_help_and_exit_error
+fi
+
+if [ ${#images[@]} -lt 1 ]; then
+    echo "No images specified. Aborting."
+    show_help_and_exit_error
+fi
+
+temp_dir=$(mktemp -d)
```

Comment:
Nit, not really an issue, but I would quote this.

Reply:
sure. quoted.